# Profitable App Profiles for the App Store and Google Play Markets
---
This project is for a company that builds Android and iOS mobile apps. All the apps are free to download with in-app purchase. 

Our goal for this project is to analyze data to help our developers understand what type of apps are likely to attract more users.

In [48]:
from csv import reader

## Read in and explore data

In [49]:
# Create a function to read open and read csv files
def read_csv(csv):
    opened_file = open(csv, encoding="utf8")
    read_file = reader(opened_file)
    return list(read_file)

In [50]:
# Read in datasets
apple_store = read_csv('AppleStore.csv')
apple_header = apple_store[0]
apple_data = apple_store[1:]

google_store = read_csv('googleplaystore.csv')
google_header = google_store[0]
google_data = google_store[1:]

In [51]:
# Create a function to print out selected rows and data structure info
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [52]:
# Explore apple_store dataset
print(apple_header)
print('*'*80)
explore_data(apple_data, 0, 5, rows_and_columns=True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']
********************************************************************************
['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '3

In [53]:
# Explore google_store dataset
print(google_header)
print('*'*80)
explore_data(google_data, 0, 5, rows_and_columns=True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
********************************************************************************
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Colori

In [54]:
# Print columns of the datasets
print(apple_store[0])
print('\n')
print(google_store[0])

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


## Data cleaning

In [55]:
# Locate the error in a certain row mentioned in the dataset comments
print(google_data[10472])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In [56]:
# Delete the row with error
del google_data[10472]

### Check for duplicated data and keep only the most recent -- the ones with the most number of reviews

In [57]:
# Check for duplicates in dataset and print the first five
"""
Since in both datasets the column to check for duplicates are both happend to be the first column
we can use one function to check for duplicates in both datasets.
"""
def get_duplicates(dataset): 
    unique_data = []
    duplicates = []
    for app in dataset:
        name = app[0]
        if name in unique_data:
            duplicates.append(name)
        else:
            unique_data.append(name)
    print('There are ', len(duplicates),' duplicates in the dataset')
    if len(duplicates)!=0:
        print('Showing first 5: ')
        [print(name) for name in duplicates[:5]]
    return duplicates, unique_data

In [58]:
# Check for duplicates in google_data
google_duplicates, google_unique = get_duplicates(google_data) #

There are  1181  duplicates in the dataset
Showing first 5: 
Quick PDF Scanner + OCR FREE
Box
Google My Business
ZOOM Cloud Meetings
join.me - Simple Meetings


In [59]:
# Check for duplicates in apple_data 
apple_duplicates = get_duplicates(apple_data)

There are  0  duplicates in the dataset


In [60]:
### Remove the duplicate entries and only keep one with most number of reviews ###

# Create a dictionary with unique app names and review number -1 for reference
reviews_max = {name:-1 for name in google_unique}

# Find apps with most number of reviews
for app in google_data:
    name = app[0]
    review_num = int(app[3])
    if review_num > reviews_max[name]:
        reviews_max[name] = review_num
        
google_clean = [] # Initialize a list to add unique apps
already_added = [] # Initialize a list for reference of apps with same name and number of reviews

# Find apps with the same name and number of reviews as reviews_max dictionary
for app in google_data:
    name = app[0]
    review_num = int(app[3])
    if review_num == reviews_max[name] and name not in already_added:
        google_clean.append(app)
        already_added.append(name)

In [61]:
# Sanity check data length 
print('Expected length: ', len(google_unique))
print('Actual length: ', len(google_clean))

Expected length:  9659
Actual length:  9659


### Keep English apps only
---
We'd like to analyze only the apps that are directed toward an English-speaking audience. However, both data sets have apps with names that suggest they are not directed toward an English-speaking audience.
<br><br>
The numbers corresponding to the characters we commonly use in an English text are all in the range 0 to 127, according to the ASCII (American Standard Code for Information Interchange) system. There are exceptions like emojis and ™ signs, which has as digital code outside of 0 to 127, but are used in English apps. 

In [62]:
# Filter out non-English android apps (apps with more than three letters outside the ASCII range) 
google_clean_eng = []

for app in google_clean:
    name = app[0]
    letters = [ord(l)>127 for l in name]
    if sum(letters) < 4:
        google_clean_eng.append(app)

In [63]:
print(google_header)
print('*'*80)
explore_data(google_clean_eng, 0, 5, rows_and_columns=True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
********************************************************************************
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instru

In [64]:
# Filter out non-English iOS apps (apps with more than three letters outside the ASCII range) 
apple_clean_eng = []

for app in apple_data:
    name = app[1]
    letters = [ord(l)>127 for l in name]
    if sum(letters) < 4:
        apple_clean_eng.append(app)

In [65]:
print(apple_header)
print('*'*80)
explore_data(apple_clean_eng, 0, 5, rows_and_columns=True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']
********************************************************************************
['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '3

### Isolate free apps

In [66]:
# Isolate free android apps 
google_free = []

for app in google_clean_eng:
    price_type = app[6]
    if price_type == 'Free':
        google_free.append(app)

In [67]:
print(google_header)
print('*'*80)
explore_data(google_free, 0, 5, rows_and_columns=True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
********************************************************************************
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instru

In [68]:
# Isolate free iOS apps
apple_free = []

for app in apple_clean_eng:
    price = app[4]
    if price == '0.0':
        apple_free.append(app)

In [69]:
print(apple_header)
print('*'*80)
explore_data(apple_free, 0, 5, rows_and_columns=True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']
********************************************************************************
['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '3

## Analyze free iOS and Android apps 
---
Our end goal is to develope apps for both Google Play and the App Store, we need to find app profiles that are successful on both markets. 

### We wiil start by getting a sense of what are the most common genres for each market. 
* In the apple_free dataset, the column `prime_genre` is the column we can use to group apps by genre
* In the google_free dataset, the column `Category` and `Genres` can both be used to group Android apps by genre.

In [70]:
# Create a function that returns a frequency table for desired column
def freq_table(dataset, col_idx):
    freq_table = {}
    for data in dataset:
        key = data[col_idx]
        if key in freq_table:
            freq_table[key]+=1
        else:
            freq_table[key] = 1
            
    # Sort frequency table by value in reverse order       
    sorted_freq_table =  {k: v for k, v in sorted(freq_table.items(), 
                                                  key = lambda item: item[1], # sort by value
                                                  reverse = True)}
    
    # Create a frequency table in fraction
    pct_freq_table = {k:'{}%'.format(round((v/len(dataset))*100,2))
                      for k, v in sorted_freq_table.items()}
    
    print('Frequency table','\n','*'*40)
    print(sorted_freq_table)
    print('\nFrequency table in %','\n','*'*40)
    print(pct_freq_table)
    
    return sorted_freq_table, pct_freq_table

In [71]:
# Display iOS app `prime_genre`  frequency table 
apple_genre_freq, apple_genre_freq_pct = freq_table(apple_free, 11)

Frequency table 
 ****************************************
{'Games': 1874, 'Entertainment': 254, 'Photo & Video': 160, 'Education': 118, 'Social Networking': 106, 'Shopping': 84, 'Utilities': 81, 'Sports': 69, 'Music': 66, 'Health & Fitness': 65, 'Productivity': 56, 'Lifestyle': 51, 'News': 43, 'Travel': 40, 'Finance': 36, 'Weather': 28, 'Food & Drink': 26, 'Reference': 18, 'Business': 17, 'Book': 14, 'Navigation': 6, 'Medical': 6, 'Catalogs': 4}

Frequency table in % 
 ****************************************
{'Games': '58.16%', 'Entertainment': '7.88%', 'Photo & Video': '4.97%', 'Education': '3.66%', 'Social Networking': '3.29%', 'Shopping': '2.61%', 'Utilities': '2.51%', 'Sports': '2.14%', 'Music': '2.05%', 'Health & Fitness': '2.02%', 'Productivity': '1.74%', 'Lifestyle': '1.58%', 'News': '1.33%', 'Travel': '1.24%', 'Finance': '1.12%', 'Weather': '0.87%', 'Food & Drink': '0.81%', 'Reference': '0.56%', 'Business': '0.53%', 'Book': '0.43%', 'Navigation': '0.19%', 'Medical': '0.19%', 

#### Observations on iOS apps genre:
* The most common genre in apple store is ***Games***, which contains around 58% of all apps. With the runner-up being ***Entertainment***, which contains around 8% of all apps. 
* All top three genres are more for entertainment and takes up around 71% of all apps. 
* Among the more practical generes, the most common one is ***Education***, which takes up 3.66% of the apps. Followed by ***Shopping***, which takes up around 2.6% of the apps.

In [72]:
# Display Android app `Genres`  frequency table 
google_genre_freq, google_genre_freq_pct = freq_table(google_free, 9);

Frequency table 
 ****************************************
{'Tools': 749, 'Entertainment': 538, 'Education': 474, 'Business': 407, 'Lifestyle': 345, 'Productivity': 345, 'Finance': 328, 'Medical': 313, 'Sports': 307, 'Personalization': 294, 'Communication': 287, 'Action': 275, 'Health & Fitness': 273, 'Photography': 261, 'News & Magazines': 248, 'Social': 236, 'Travel & Local': 206, 'Shopping': 199, 'Books & Reference': 190, 'Simulation': 181, 'Dating': 165, 'Arcade': 164, 'Video Players & Editors': 157, 'Casual': 156, 'Maps & Navigation': 124, 'Food & Drink': 110, 'Puzzle': 100, 'Racing': 88, 'Libraries & Demo': 83, 'Role Playing': 83, 'Auto & Vehicles': 82, 'Strategy': 80, 'House & Home': 73, 'Weather': 71, 'Events': 63, 'Adventure': 60, 'Comics': 54, 'Art & Design': 53, 'Beauty': 53, 'Parenting': 44, 'Card': 40, 'Casino': 38, 'Trivia': 37, 'Educational;Education': 35, 'Board': 34, 'Educational': 33, 'Education;Education': 30, 'Word': 23, 'Casual;Pretend Play': 21, 'Music': 18, 'Ente

In [73]:
# Display Android app `Category` frequency table 
google_cat_freq, google_cat_freq_pct = freq_table(google_free, 1)

Frequency table 
 ****************************************
{'FAMILY': 1675, 'GAME': 862, 'TOOLS': 750, 'BUSINESS': 407, 'LIFESTYLE': 346, 'PRODUCTIVITY': 345, 'FINANCE': 328, 'MEDICAL': 313, 'SPORTS': 301, 'PERSONALIZATION': 294, 'COMMUNICATION': 287, 'HEALTH_AND_FITNESS': 273, 'PHOTOGRAPHY': 261, 'NEWS_AND_MAGAZINES': 248, 'SOCIAL': 236, 'TRAVEL_AND_LOCAL': 207, 'SHOPPING': 199, 'BOOKS_AND_REFERENCE': 190, 'DATING': 165, 'VIDEO_PLAYERS': 159, 'MAPS_AND_NAVIGATION': 124, 'FOOD_AND_DRINK': 110, 'EDUCATION': 103, 'ENTERTAINMENT': 85, 'LIBRARIES_AND_DEMO': 83, 'AUTO_AND_VEHICLES': 82, 'HOUSE_AND_HOME': 73, 'WEATHER': 71, 'EVENTS': 63, 'PARENTING': 58, 'ART_AND_DESIGN': 57, 'COMICS': 55, 'BEAUTY': 53}

Frequency table in % 
 ****************************************
{'FAMILY': '18.9%', 'GAME': '9.73%', 'TOOLS': '8.46%', 'BUSINESS': '4.59%', 'LIFESTYLE': '3.9%', 'PRODUCTIVITY': '3.89%', 'FINANCE': '3.7%', 'MEDICAL': '3.53%', 'SPORTS': '3.4%', 'PERSONALIZATION': '3.32%', 'COMMUNICATION': '3.2

In [74]:
# Since the category 'FAMILY' is ambiguous, let's check out apps that belongs to this category
google_family_apps = []
for app in google_free:
    cat = app[1]
    if cat == 'FAMILY':
        google_family_apps.append(app)
        
# Check out more detailed genres in google family apps   
freq_table(google_family_apps, 9);

Frequency table 
 ****************************************
{'Entertainment': 458, 'Education': 382, 'Simulation': 174, 'Casual': 134, 'Puzzle': 78, 'Role Playing': 72, 'Strategy': 65, 'Educational;Education': 35, 'Educational': 33, 'Education;Education': 24, 'Casual;Pretend Play': 21, 'Puzzle;Brain Games': 15, 'Racing;Action & Adventure': 15, 'Casual;Action & Adventure': 12, 'Entertainment;Music & Video': 12, 'Casual;Brain Games': 11, 'Arcade;Action & Adventure': 11, 'Educational;Pretend Play': 8, 'Action;Action & Adventure': 8, 'Simulation;Action & Adventure': 7, 'Board;Brain Games': 7, 'Educational;Brain Games': 6, 'Entertainment;Brain Games': 6, 'Casual;Creativity': 6, 'Education;Pretend Play': 4, 'Role Playing;Pretend Play': 4, 'Educational;Creativity': 3, 'Adventure;Action & Adventure': 3, 'Role Playing;Action & Adventure': 3, 'Educational;Action & Adventure': 3, 'Entertainment;Action & Adventure': 3, 'Puzzle;Action & Adventure': 3, 'Education;Action & Adventure': 3, 'Education;Mu

#### Observations on Android apps category:
* The `Genres` column in Android apps is quite scattered, the `Category` column seems more comparable with iOS apps genre. So we will use that as a main source of observation. 
* The most common category in Google play store is ***FAMILY***, which contains around 19% of all apps. With the runner-up being ***GAME***, which contains around 10% of all apps. 
* In the top 5 categories, only ***GAME*** is clearly more for entertainment. ***TOOLS, BUSINESS, LIFESTYLE*** are clearly more for practicle purposes, and ***FAMILY*** is ambiguous as in which type it belongs to. 
* The most common genres in Google play store is ***Tools***, which contains around 8.45% of all apps'. With the runner-up being ***Entertainment***, which contains around 6.07% of all apps. Followed by ***Education***, which contains around 5.35% of all apps.
* In the top 5 genres, only ***Entertainment*** is clearly for entertainment. ***Tools, Education, Business, Lifestyle*** are clearly more for practicle purposes.

---
#### Conclusion:
* Unlike generes in apple store, categories and genres in Google play store are much less skewed. The most common genres in apple store are more for entertainment, while in google play store it's more balanced between entertainment apps and apps for practicle purposes. 
* Based on the observations in both markets, we can see that the ***GAME*** genre is clearly more common in both markets. ***Entertainment*** and ***Education*** can both be strong contenders too, based on the more detailed genre in ***FAMILY*** category in Google play store. 
* Based on the freqency tables, we can get a sense of which genres are more common. Although, just because a genere is more common, doesn't necessarily mean it's more popular or has more users in general. We will need more data, like number of installs and user reviews, to varify our guesses.

### Analyze popularity of genres among users in App store and Google play store
---
One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. <br><br>For the Google Play data set, we can find this information in the Installs column, but this information is missing for the App Store data set. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the `rating_count_tot` column.

### iOS app popularity by genre

In [75]:
# Calculate average rating count of each genre for iOS apps
apple_rating_cnt_tot = {genre:0 for genre in apple_genre_freq}

# Get total rating count for each genre
for app in apple_free:
    rating_cnt = int(app[5])  
    genre = app[11]
    apple_rating_cnt_tot[genre]+=rating_cnt

# Get average rating count for each genre
apple_rating_cnt_avg = {genre:apple_rating_cnt_tot[genre]/apple_genre_freq[genre] for genre in apple_genre_freq}

# Sort by descending order 
apple_rating_cnt_avg = {genre:avg_rating for genre, avg_rating in sorted(apple_rating_cnt_avg.items(), 
                                                  key = lambda item: item[1], reverse = True)} 
apple_rating_cnt_avg

{'Navigation': 86090.33333333333,
 'Reference': 74942.11111111111,
 'Social Networking': 71548.34905660378,
 'Music': 57326.530303030304,
 'Weather': 52279.892857142855,
 'Book': 39758.5,
 'Food & Drink': 33333.92307692308,
 'Finance': 31467.944444444445,
 'Photo & Video': 28441.54375,
 'Travel': 28243.8,
 'Shopping': 26919.690476190477,
 'Health & Fitness': 23298.015384615384,
 'Sports': 23008.898550724636,
 'Games': 22788.6696905016,
 'News': 21248.023255813954,
 'Productivity': 21028.410714285714,
 'Utilities': 18684.456790123455,
 'Lifestyle': 16485.764705882353,
 'Entertainment': 14029.830708661417,
 'Business': 7491.117647058823,
 'Education': 7003.983050847458,
 'Catalogs': 4004.0,
 'Medical': 612.0}

In [76]:
# Explore the number of apps in each genre that has a higher than average rating count

for genre, avg_rating_cnt in apple_rating_cnt_avg.items():
    above_avg_cnt = 0
    below_avg_cnt = 0
    total_num = apple_genre_freq[genre]
    for app in apple_free:
        rating_cnt = float(app[5])
        g = app[11]
        if g == genre and rating_cnt > avg_rating_cnt:
            above_avg_cnt+=1
        elif g == genre and rating_cnt < avg_rating_cnt:
            below_avg_cnt+=1
            
    print('In *{}* genre:\n'.format(genre))
    print('{} apps({}%) have above average rating count.\n'.format(above_avg_cnt, 
                                                                     round((above_avg_cnt/total_num)*100,2)))
    print('{} apps({}%) have below average rating count.\n'.format(below_avg_cnt, 
                                                                     round((below_avg_cnt/total_num)*100,2)))
    print('*'*50, '\n')

In *Navigation* genre:

2 apps(33.33%) have above average rating count.

4 apps(66.67%) have below average rating count.

************************************************** 

In *Reference* genre:

2 apps(11.11%) have above average rating count.

16 apps(88.89%) have below average rating count.

************************************************** 

In *Social Networking* genre:

16 apps(15.09%) have above average rating count.

90 apps(84.91%) have below average rating count.

************************************************** 

In *Music* genre:

10 apps(15.15%) have above average rating count.

56 apps(84.85%) have below average rating count.

************************************************** 

In *Weather* genre:

6 apps(21.43%) have above average rating count.

22 apps(78.57%) have below average rating count.

************************************************** 

In *Book* genre:

5 apps(35.71%) have above average rating count.

9 apps(64.29%) have below average rating count.

*****

### Observations:
Now let's be realistic. While ***Navigation*** has the highest average rating count, it also has only 6 apps in total and 2 apps have above average rating count. This tells us that this genre is not easy to break into, and probably contains 2 apps that dominate the market. It's not hard to guess one of them is Google maps. The same logic can be applied to ***Reference*** and ***Social Networking***.
<br>

By the nature of each genre, we can also leave out:
* ***Weather***, which people don't usually spend a lot of time on.
* ***Food & Drink***, ***Catalogs*** and ***Shopping*** which are usually based on/requires offline business.
* ***Medical***, and ***Finance***, which requires specialized knowledge.
* ***Music***, ***Book*** and ***News***, which requires content creation or publishing rights. For these genres, the real workload isn't app development.

That leaves us with the following genres: 

|       Genre      | Average Rating Count | Above Average% | Total App Number |
|:----------------:|:--------------------:|:--------------:|:----------------:|
|   Photo & Video  |         28442        |      14.37     |        160       |
|      Travel      |         28244        |      22.5      |        40        |
| Health & Fitness |         23298        |      12.31     |        65        |
|      Sports      |         23009        |      20.29     |        69        |
|       Games      |         22789        |      14.94     |       1874       |
|   Productivity   |         21028        |      26.79     |        56        |
|     Utilities    |         18684        |      19.75     |        81        |
|     Lifestyle    |         16486        |      15.69     |        51        |
|   Entertainment  |         14030        |      18.9      |        254       |
|     Business     |         7491         |      41.18     |        17        |
|     Education    |         7003         |      16.95     |        118       |

Even though ***Photo & Video*** is on the top of the list, the apps with above average percentage is low, which means for most apps in this genre, their rating count are below average. With that in mind, the runner up genre ***Travel*** has a very close average rating count and a higher above average percentage. It doesn't have a lot of apps, which means less competetion. And it has a highest percentage of apps that has above average rating count in top 5 genres, which means the average rating count is more representitive. Let's check out the apps in this genre. 

In [118]:
# Create a function to get top n iOS apps in a specified genre and it's review count
def top_n_app_rating_cnt(top_n, compare_genre):
    i = 0
    for app in apple_free:       
        genre = app[11]
        count = int(app[5])
        name = app[1]
        if genre == compare_genre and i < top_n:
            print(name, ':', count, 'reviews')
            print('\n')
            i+=1

In [119]:
# Check out iOS apps in *Travel* genre
top_n_app_rating_cnt(top_n=10, compare_genre='Travel')

Google Earth : 446185 reviews


Yelp - Nearby Restaurants, Shopping & Services : 223885 reviews


GasBuddy : 145549 reviews


TripAdvisor Hotels Flights Restaurants : 56194 reviews


Uber : 49466 reviews


Lyft : 46922 reviews


HotelTonight - Great Deals on Last Minute Hotels : 32341 reviews


Hotels & Vacation Rentals by Booking.com : 31261 reviews


Southwest Airlines : 30552 reviews


Airbnb : 22302 reviews




Looks like the ***Travel*** genre is out too, since it's mostly travel booking apps, that requires a business, or navigation apps. 
<br>
Let's check out ***Health & Fitness***, ***Sports*** and ***Productivity*** genre next. 

In [120]:
# Check out iOS apps in *Health & Fitness* genre
top_n_app_rating_cnt(top_n=10, compare_genre='Health & Fitness')

Calorie Counter & Diet Tracker by MyFitnessPal : 507706 reviews


Lose It! – Weight Loss Program and Calorie Counter : 373835 reviews


Weight Watchers : 136833 reviews


Sleep Cycle alarm clock : 104539 reviews


Fitbit : 90496 reviews


Period Tracker Lite : 53620 reviews


Nike+ Training Club - Workouts & Fitness Plans : 33969 reviews


Plant Nanny - Water Reminder with Cute Plants : 27421 reviews


Sworkit - Custom Workouts for Exercise & Fitness : 16819 reviews


Clue Period Tracker: Period & Ovulation Tracker : 13436 reviews




In [122]:
# Check out iOS apps in *Sports* genre
top_n_app_rating_cnt(top_n=10, compare_genre='Sports')

ESPN: Get scores, news, alerts & watch live sports : 290996 reviews


Yahoo Fantasy Sports : 190670 reviews


WatchESPN : 159735 reviews


The Masters Tournament : 148160 reviews


Yahoo Sports - Teams, Scores, News & Highlights : 137951 reviews


ESPN Fantasy Football Baseball Basketball Hockey : 64925 reviews


CBS Sports App - Sports Scores, News, Stats, Watch : 59639 reviews


FOX Sports Mobile : 57500 reviews


2016 U.S. Open Golf Championship : 54192 reviews


NBC Sports : 47172 reviews




In [123]:
# Check out iOS apps in *Productivity* genre
top_n_app_rating_cnt(top_n=10, compare_genre='Productivity')

Evernote - stay organized : 161065 reviews


Gmail - email by Google: secure, fast & organized : 135962 reviews


iTranslate - Language Translator & Dictionary : 123215 reviews


Yahoo Mail - Keeps You Organized! : 113709 reviews


Google Docs : 64259 reviews


Google Drive - free online storage : 59255 reviews


Dropbox : 49578 reviews


Microsoft Word : 47999 reviews


Microsoft OneNote : 39638 reviews


Microsoft Outlook - email and calendar : 32807 reviews




Among ***Health & Fitness***, ***Sports*** and ***Productivity*** genres, ***Sports*** genre is dominated by sports news channels and fantasy sports. ***Productivity*** has mostly office apps. The one with some potential is the ***Health & Fitness*** genre. 
<br><br>
In ***Health & Fitness*** genre, the calorie counter type is dominated by the top apps, but there are still room in workout and exercise. For instance more niche workout apps like **yoga, self-weight, 10-minutes workouts, physical theropy workout** etc, with features that motivate users. Also inspired by period tracker, men experience hormonal cycles too, so a mood/hormonal tracker for men is an angle. Or fitness apps more customized for specific gender like diet planner based on daily needs for different gender or medical conditions. I can go on for a while...

### Android app popularity by genre
---
We have data about the number of installs for the Google Play market, so we should be able to get a clearer picture about genre popularity. However, the install numbers don't seem precise enough — we can see that most values are open-ended (100+, 1,000+, 5,000+, etc.):

In [82]:
# Check out the Installs column values 
google_installs_freq, google_installs_freq_pct = freq_table(google_free, 5)

Frequency table 
 ****************************************
{'1,000,000+': 1394, '100,000+': 1024, '10,000,000+': 935, '10,000+': 904, '1,000+': 744, '100+': 613, '5,000,000+': 605, '500,000+': 493, '50,000+': 423, '5,000+': 400, '10+': 314, '500+': 288, '50,000,000+': 204, '100,000,000+': 189, '50+': 170, '5+': 70, '1+': 45, '500,000,000+': 24, '1,000,000,000+': 20, '0+': 4}

Frequency table in % 
 ****************************************
{'1,000,000+': '15.73%', '100,000+': '11.55%', '10,000,000+': '10.55%', '10,000+': '10.2%', '1,000+': '8.39%', '100+': '6.92%', '5,000,000+': '6.83%', '500,000+': '5.56%', '50,000+': '4.77%', '5,000+': '4.51%', '10+': '3.54%', '500+': '3.25%', '50,000,000+': '2.3%', '100,000,000+': '2.13%', '50+': '1.92%', '5+': '0.79%', '1+': '0.51%', '500,000,000+': '0.27%', '1,000,000,000+': '0.23%', '0+': '0.05%'}


We don't need very precise data for our purposes — we only want to find out which app genres attract the most users, and we don't need perfect precision with respect to the number of users. We're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on.

In [83]:
import re 
# Create a new list of apps with installs converted to int and exclude irrelevant columns
google_free_installs = []

for app in google_free:
    name = app[0]
    cat = app[1]
    genre = app[9]
    install = app[5]
    new_app = [name, cat, genre, install]
    new_app[3] = int(re.sub(r'\D*', '', install)) # Strip all non-digits from the intalls and convert to int
    google_free_installs.append(new_app)

In [84]:
print(['name', 'category', 'genre', 'installs'], '\n')
explore_data(google_free_installs, 0, 5)

['name', 'category', 'genre', 'installs'] 

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', 'Art & Design', 10000]


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', 'Art & Design', 5000000]


['Sketch - Draw & Paint', 'ART_AND_DESIGN', 'Art & Design', 50000000]


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', 'Art & Design;Creativity', 100000]


['Paper flowers instructions', 'ART_AND_DESIGN', 'Art & Design', 50000]




In [86]:
# Create a dictionary that contains each category and the total installs 
google_cat_install_tot = {cat:0 for cat in google_cat_freq}

for app in google_free_installs:
    cat = app[1]
    install = app[3]
    google_cat_install_tot[cat]+=install

# Get average install for each category
google_cat_install_avg = {cat:google_cat_install_tot[cat]/google_cat_freq[cat] for cat in google_cat_freq}

# Sort category by install in descending order
google_cat_install_avg = {cat:avg_install for cat, avg_install in sorted(google_cat_install_avg.items(), 
                                                                       key = lambda item: item[1],
                                                                      reverse = True)}

In [87]:
google_cat_install_avg

{'COMMUNICATION': 38456119.167247385,
 'VIDEO_PLAYERS': 24727872.452830188,
 'SOCIAL': 23253652.127118643,
 'PHOTOGRAPHY': 17840110.40229885,
 'PRODUCTIVITY': 16787331.344927534,
 'GAME': 15588015.603248259,
 'TRAVEL_AND_LOCAL': 13984077.710144928,
 'ENTERTAINMENT': 11640705.88235294,
 'TOOLS': 10801391.298666667,
 'NEWS_AND_MAGAZINES': 9549178.467741935,
 'BOOKS_AND_REFERENCE': 8767811.894736841,
 'SHOPPING': 7036877.311557789,
 'PERSONALIZATION': 5201482.6122448975,
 'WEATHER': 5074486.197183099,
 'HEALTH_AND_FITNESS': 4188821.9853479853,
 'MAPS_AND_NAVIGATION': 4056941.7741935486,
 'FAMILY': 3697848.1731343283,
 'SPORTS': 3638640.1428571427,
 'ART_AND_DESIGN': 1986335.0877192982,
 'FOOD_AND_DRINK': 1924897.7363636363,
 'EDUCATION': 1833495.145631068,
 'BUSINESS': 1712290.1474201474,
 'LIFESTYLE': 1437816.2687861272,
 'FINANCE': 1387692.475609756,
 'HOUSE_AND_HOME': 1331540.5616438356,
 'DATING': 854028.8303030303,
 'COMICS': 817657.2727272727,
 'AUTO_AND_VEHICLES': 647317.8170731707

From the experience with analyzing iOS apps, we can guess that in some categories, there will be a few top apps that are outliers and make the average installs misleading.
<br><br>
Since in Android apps the installs are rough numbers, we can easily get how many apps are in each install range. What we will do next is to calculate the numbers of apps in each category in each unique install. 

In [93]:
# Extract unique installs 
unique_installs = set(app[3] for app in google_free_installs)
unique_installs

{0,
 1,
 5,
 10,
 50,
 100,
 500,
 1000,
 5000,
 10000,
 50000,
 100000,
 500000,
 1000000,
 5000000,
 10000000,
 50000000,
 100000000,
 500000000,
 1000000000}

In [88]:
# Sort google installs frequency table by installs
google_installs_freq_pct = {installs:count for installs, count in sorted(google_installs_freq_pct.items(),
                                                                    key = lambda item: len(item[0]),
                                                                    reverse = True)}
google_installs_freq_pct

{'1,000,000,000+': '0.23%',
 '100,000,000+': '2.13%',
 '500,000,000+': '0.27%',
 '10,000,000+': '10.55%',
 '50,000,000+': '2.3%',
 '1,000,000+': '15.73%',
 '5,000,000+': '6.83%',
 '100,000+': '11.55%',
 '500,000+': '5.56%',
 '10,000+': '10.2%',
 '50,000+': '4.77%',
 '1,000+': '8.39%',
 '5,000+': '4.51%',
 '100+': '6.92%',
 '500+': '3.25%',
 '10+': '3.54%',
 '50+': '1.92%',
 '5+': '0.79%',
 '1+': '0.51%',
 '0+': '0.05%'}

From the installs frequency table in percentage, let's group the installs range into less granularity. If group by roughly every 20%-35%, we will get these groups: less than 5,000, between [5,000, 500,000], between [500,000, 1,000,000] and 1,000,000+

In [89]:
# Create a list of installs groups
installs_range = ['Less than 5000', '5,000-500,000', '500,000-1,000,000', '1,000,000+']

In [90]:
# Create a list of categories as column
cat_install_crosstab = {cat:None for cat in google_cat_install_avg}
cat_install_crosstab

{'COMMUNICATION': None,
 'VIDEO_PLAYERS': None,
 'SOCIAL': None,
 'PHOTOGRAPHY': None,
 'PRODUCTIVITY': None,
 'GAME': None,
 'TRAVEL_AND_LOCAL': None,
 'ENTERTAINMENT': None,
 'TOOLS': None,
 'NEWS_AND_MAGAZINES': None,
 'BOOKS_AND_REFERENCE': None,
 'SHOPPING': None,
 'PERSONALIZATION': None,
 'WEATHER': None,
 'HEALTH_AND_FITNESS': None,
 'MAPS_AND_NAVIGATION': None,
 'FAMILY': None,
 'SPORTS': None,
 'ART_AND_DESIGN': None,
 'FOOD_AND_DRINK': None,
 'EDUCATION': None,
 'BUSINESS': None,
 'LIFESTYLE': None,
 'FINANCE': None,
 'HOUSE_AND_HOME': None,
 'DATING': None,
 'COMICS': None,
 'AUTO_AND_VEHICLES': None,
 'LIBRARIES_AND_DEMO': None,
 'PARENTING': None,
 'BEAUTY': None,
 'EVENTS': None,
 'MEDICAL': None}

In [94]:
for category in cat_install_crosstab:
    # Initialize a dictionary with unique_installs and zeros as counter
    install_count = {installs:0 for installs in installs_range}
    print(category)
    for app in google_free_installs:
        name = app[0]
        cat = app[1]
        installs = app[3]
        if cat == category:
            if installs >= 1000000:
                install_count['1,000,000+']+=1
            elif 500000 <= installs < 1000000:
                install_count['500,000-1,000,000']+=1
            elif 5000 <= installs < 500000:
                install_count['5,000-500,000']+=1
            else:
                install_count['Less than 5000']+=1
    # Sort install_count by installs in descending order and assign to value 
    
    cat_install_crosstab[category] = install_count

COMMUNICATION
VIDEO_PLAYERS
SOCIAL
PHOTOGRAPHY
PRODUCTIVITY
GAME
TRAVEL_AND_LOCAL
ENTERTAINMENT
TOOLS
NEWS_AND_MAGAZINES
BOOKS_AND_REFERENCE
SHOPPING
PERSONALIZATION
WEATHER
HEALTH_AND_FITNESS
MAPS_AND_NAVIGATION
FAMILY
SPORTS
ART_AND_DESIGN
FOOD_AND_DRINK
EDUCATION
BUSINESS
LIFESTYLE
FINANCE
HOUSE_AND_HOME
DATING
COMICS
AUTO_AND_VEHICLES
LIBRARIES_AND_DEMO
PARENTING
BEAUTY
EVENTS
MEDICAL


In [95]:
cat_install_crosstab

{'COMMUNICATION': {'Less than 5000': 77,
  '5,000-500,000': 62,
  '500,000-1,000,000': 9,
  '1,000,000+': 139},
 'VIDEO_PLAYERS': {'Less than 5000': 23,
  '5,000-500,000': 47,
  '500,000-1,000,000': 4,
  '1,000,000+': 85},
 'SOCIAL': {'Less than 5000': 62,
  '5,000-500,000': 60,
  '500,000-1,000,000': 14,
  '1,000,000+': 100},
 'PHOTOGRAPHY': {'Less than 5000': 36,
  '5,000-500,000': 41,
  '500,000-1,000,000': 12,
  '1,000,000+': 172},
 'PRODUCTIVITY': {'Less than 5000': 109,
  '5,000-500,000': 76,
  '500,000-1,000,000': 13,
  '1,000,000+': 147},
 'GAME': {'Less than 5000': 81,
  '5,000-500,000': 189,
  '500,000-1,000,000': 61,
  '1,000,000+': 531},
 'TRAVEL_AND_LOCAL': {'Less than 5000': 47,
  '5,000-500,000': 58,
  '500,000-1,000,000': 9,
  '1,000,000+': 93},
 'ENTERTAINMENT': {'Less than 5000': 0,
  '5,000-500,000': 11,
  '500,000-1,000,000': 1,
  '1,000,000+': 73},
 'TOOLS': {'Less than 5000': 180,
  '5,000-500,000': 270,
  '500,000-1,000,000': 26,
  '1,000,000+': 274},
 'NEWS_AND_

After regrouping, we can see that ***Communication*** no longer holds the fisrt place in number of apps with over 1,000,000 installs. From which we can deduce that there are probably few outliers in the ***Communication*** category that have significantly higher installs. 

---
The top 5 categories with most numbers of apps that have 1,000,000 installs are now:


|   Category   | # Apps with 1,000,000+ installs |
|:------------:|:-----------------------------:|
|    FAMILY    |              557              |
|     GAME     |              531              |
|     TOOLS    |              274              |
|  PHOTOGRAPHY |              172              |
| PRODUCTIVITY |              147              |

We have checked the ***FAMILY*** category in detail before: 'Entertainment': '27.34%', 'Education': '22.81%', 'Simulation': '10.39%', 'Casual': '8.0%', takes up almost 70% of this category. 
<br><br>
Let's checkout the top 10 apps in these categories.

In [99]:
# Sort google_free_installs by installs in descending order
sorted_google_installs = sorted(google_free_installs, key = lambda app: app[3], reverse = True)
sorted_google_installs[:5]

[['Google Play Books', 'BOOKS_AND_REFERENCE', 'Books & Reference', 1000000000],
 ['WhatsApp Messenger', 'COMMUNICATION', 'Communication', 1000000000],
 ['Messenger – Text and Video Chat for Free',
  'COMMUNICATION',
  'Communication',
  1000000000],
 ['Skype - free IM & video calls',
  'COMMUNICATION',
  'Communication',
  1000000000],
 ['Google Chrome: Fast & Secure',
  'COMMUNICATION',
  'Communication',
  1000000000]]

In [109]:
# Create a function that returns the top 10 android apps in a category
def top_10(cat):
    i = 0
    print('In {} category:\n'.format(cat))
    for app in sorted_google_installs:
        name = app[0]
        category = app[1]
        genre = app[2]
        installs = app[3]
        if category == cat and i < 10:
            print("""{} in genre {} has {} installs.""".format(name, genre, installs))
            print('\n')
            i+=1

In [110]:
top_10('FAMILY')

In FAMILY category:

Google Play Games in genre Entertainment has 1000000000 installs.


Minion Rush: Despicable Me Official Game in genre Casual;Action & Adventure has 100000000 installs.


ROBLOX in genre Adventure;Action & Adventure has 100000000 installs.


Hay Day in genre Casual has 100000000 installs.


Talking Tom Cat 2 in genre Entertainment has 100000000 installs.


Talking Ginger in genre Entertainment has 100000000 installs.


Where's My Water? Free in genre Puzzle;Brain Games has 100000000 installs.


Talking Tom Cat in genre Casual has 100000000 installs.


Tom Loves Angela in genre Entertainment has 100000000 installs.


Talking Tom & Ben News in genre Entertainment has 100000000 installs.




In [111]:
top_10('GAME')

In GAME category:

Subway Surfers in genre Arcade has 1000000000 installs.


Candy Crush Saga in genre Casual has 500000000 installs.


Pou in genre Casual has 500000000 installs.


My Talking Tom in genre Casual has 500000000 installs.


Temple Run 2 in genre Action has 500000000 installs.


Sonic Dash in genre Arcade has 100000000 installs.


PAC-MAN in genre Arcade has 100000000 installs.


Roll the Ball® - slide puzzle in genre Puzzle has 100000000 installs.


Piano Tiles 2™ in genre Arcade has 100000000 installs.


Pokémon GO in genre Adventure has 100000000 installs.




In [112]:
top_10('TOOLS')

In TOOLS category:

Google in genre Tools has 1000000000 installs.


Google Translate in genre Tools has 500000000 installs.


SHAREit - Transfer & Share in genre Tools has 500000000 installs.


Gboard - the Google Keyboard in genre Tools has 500000000 installs.


Clean Master- Space Cleaner & Antivirus in genre Tools has 500000000 installs.


Security Master - Antivirus, VPN, AppLock, Booster in genre Tools has 500000000 installs.


Calculator in genre Tools has 100000000 installs.


Device Help in genre Tools has 100000000 installs.


Account Manager in genre Tools has 100000000 installs.


Samsung Calculator in genre Tools has 100000000 installs.




In [114]:
top_10('PHOTOGRAPHY')

In PHOTOGRAPHY category:

Google Photos in genre Photography has 1000000000 installs.


B612 - Beauty & Filter Camera in genre Photography has 100000000 installs.


YouCam Makeup - Magic Selfie Makeovers in genre Photography has 100000000 installs.


Sweet Selfie - selfie camera, beauty cam, photo edit in genre Photography has 100000000 installs.


Retrica in genre Photography has 100000000 installs.


Photo Editor Pro in genre Photography has 100000000 installs.


BeautyPlus - Easy Photo Editor & Selfie Camera in genre Photography has 100000000 installs.


PicsArt Photo Studio: Collage Maker & Pic Editor in genre Photography has 100000000 installs.


Photo Collage Editor in genre Photography has 100000000 installs.


Z Camera - Photo Editor, Beauty Selfie, Collage in genre Photography has 100000000 installs.




In [113]:
top_10('PRODUCTIVITY')

In PRODUCTIVITY category:

Google Drive in genre Productivity has 1000000000 installs.


Microsoft Word in genre Productivity has 500000000 installs.


Dropbox in genre Productivity has 500000000 installs.


Google Calendar in genre Productivity has 500000000 installs.


Cloud Print in genre Productivity has 500000000 installs.


Microsoft Outlook in genre Productivity has 100000000 installs.


Microsoft OneDrive in genre Productivity has 100000000 installs.


Microsoft OneNote in genre Productivity has 100000000 installs.


Google Keep in genre Productivity has 100000000 installs.


ES File Explorer File Manager in genre Productivity has 100000000 installs.




Looking through the top 10 apps in these categories, the most popular apps in ***FAMILY*** are actually GAMES. Other than that, the category ***PHOTOGRAPHY*** catches our eye because it's also on the list of potentially popular iOS genres. We passed it up before because the runner up was really close and has a higer above average percentage. But since it's also a populuar category in Android, let's go back and check out this genre in iOS apps. 

In [127]:
# Check out top 15 iOS apps in Photo & Video genre
top_n_app_rating_cnt(top_n=15, compare_genre='Photo & Video')

Instagram : 2161558 reviews


Snapchat : 323905 reviews


YouTube - Watch Videos, Music, and Live Streams : 278166 reviews


Pic Collage - Picture Editor & Photo Collage Maker : 123433 reviews


Funimate video editor: add cool effects to videos : 123268 reviews


musical.ly - your video social network : 105429 reviews


Photo Collage Maker & Photo Editor - Live Collage : 93781 reviews


Vine Camera : 90355 reviews


Google Photos - unlimited photo and video storage : 88742 reviews


Flipagram : 79905 reviews


Mixgram - Picture Collage Maker - Pic Photo Editor : 54282 reviews


Shutterfly: Prints, Photo Books, Cards Made Easy : 51427 reviews


Pic Jointer – Photo Collage, Camera Effects Editor : 51330 reviews


Color Pop Effects - Photo Editor & Picture Editing : 45320 reviews


Photo Grid - photo collage maker & photo editor : 40531 reviews




## Conclusion:
While the ***Photography*** genre is kinda saturated, it's also a popular genre in both Apple store and Google play store markets. 
<br><br>
We see a lot of apps with simliar functionalities like filters, beauty cams, photo collage editor. From personal experience, there's alos a particular aspect that attracts users with photo editing apps -- the stickers. So maybe what we can develop are photo editing apps with themed stickers. For instance, holidy themed photo editors for x-mas cards to send, with printing features or e-card features. Or photo apps with movie themed stickers, apps with game themed stickers, popular pop culture themed stickers etc. Make the apps easy to share, easy to meme or doodle for more competitiveness.
<br><br>
Plus, since our goal of revenue is in-app purchase, premimum stickers or print features are very commonly paid features that users won't have a har